doc/user/basic.rst

```diff
@@ -442,6 +442,71 @@ Puppet, etc.), upgrade considerations differ somewhat:
    the same checks as when rolling out upgrades.
 
 
+.. _protocol-key-encryption:
+
+Protocol Key Encryption
+-----------------------
+
+Various routing protocols include configurable passwords, authentication
+strings, or other sorts of key strings that will be referred to in this
+section as "protocol keys." Historically, FRR has written these protocol keys
+as plain text to configuration files and to the craft interface.
+
+In some environments, it is preferable to encrypt these strings in saved
+and displayed configuration.
```

Comment:
Since we're instantiating crypto here without a prior spec:

Please describe the exact scenario(s) this is intended to address, and if there are any obvious "false positive" cases where it shouldn't be used, those too.